File: nas_spec/code_samples/Node/disputes@{dispute_id}@evidence/put.js
import { Checkout } from 'checkout-node-sdk';

let cko = new Checkout('your_client_secret_here', {
	client: 'ack_XXXXXXXXXXXX',
	scope: ['gateway'], // array of scopes
	environment: 'sandbox', // or "production"
});

// Or if you use api keys:
// const cko = new Checkout('sk_sbox_XXX', { pk: 'pk_sbox_XXX'}});

try {
	const evidence = await cko.disputes.provideEvidence('dsp_bc94ebda8d275i461229', {
		proof_of_delivery_or_service_text: 'http://checkout.com/document.pdf',
	});
} catch (err) {
	console.log(err.name);
}

File: nas_spec/code_samples/Go/disputes@{dispute_id}@evidence/put.go
// For more information please refer to https://github.com/checkout/checkout-sdk-go
import (
	"github.com/checkout/checkout-sdk-go"
	"github.com/checkout/checkout-sdk-go/configuration"
	"github.com/checkout/checkout-sdk-go/disputes"
)

// API Keys
api, err := checkout.
			Builder().
			StaticKeys().
			WithSecretKey("secret_key").
			WithEnvironment(configuration.Sandbox()). // or Environment.PRODUCTION
			Build()
if err != nil {
	return nil, err
}

// OAuth
api, err := checkout.
			Builder().
			OAuth().
			WithClientCredentials("client_id", "client_secret").
			WithScopes([]string{configuration.Disputes}).
			WithEnvironment(configuration.Sandbox()). // or Environment.PRODUCTION
			Build()
if err != nil {
	return nil, err
}

request := disputes.Evidence{
	ProofOfDeliveryOrServiceFile:           "file_id",
	ProofOfDeliveryOrServiceText:           "proof of delivery or service text",
	InvoiceOrReceiptFile:                   "file_id",
	InvoiceOrReceiptText:                   "proof of receipt text",
	InvoiceShowingDistinctTransactionsFile: "file_id",
	InvoiceShowingDistinctTransactionsText: "invoice showing distinct transactions text",
	CustomerCommunicationFile:              "file_id",
	CustomerCommunicationText:              "customer communication text",
	RefundOrCancellationPolicyFile:         "file_id",
	RefundOrCancellationPolicyText:         "refund or cancellation policy text",
	RecurringTransactionAgreementFile:      "file_id",
	RecurringTransactionAgreementText:      "recurring transaction agreement text",
	AdditionalEvidenceFile:                 "file_id",
	AdditionalEvidenceText:                 "additional evidence text",
	ProofOfDeliveryOrServiceDateFile:       "file_id",
	ProofOfDeliveryOrServiceDateText:       "proof of delivery or service date text",
}

response, err := api.Disputes.PutEvidence("dispute_id", request)
if err != nil {
	return nil, err
}

return response, nil
